api/src/opentrons/protocol_engine/commands/thermocycler/run_profile.py

```diff
@@ -0,0 +1,108 @@
+"""Command models to execute a Thermocycler profile."""
+from __future__ import annotations
+from typing import List, Optional, TYPE_CHECKING
+from typing_extensions import Literal, Type
+
+from pydantic import BaseModel, Field
+
+from ..command import AbstractCommandImpl, BaseCommand, BaseCommandCreate
+
+if TYPE_CHECKING:
+    from opentrons.protocol_engine.state import StateView
+    from opentrons.protocol_engine.execution import EquipmentHandler
+
+
+RunProfileCommandType = Literal["thermocycler/runProfile"]
+
+
+class RunProfileStepParams(BaseModel):
+    """Input parameters for an individual Thermocycler profile step."""
+
+    celsius: float = Field(..., description="Target temperature in °C.")
+    holdSeconds: int = Field(
+        ..., description="Time to hold target temperature at in seconds."
+    )
+
+
+class RunProfileParams(BaseModel):
+    """Input parameters to run a Thermocycler profile."""
+
+    moduleId: str = Field(..., description="Unique ID of the Thermocycler.")
+    profile: List[RunProfileStepParams] = Field(
+        ...,
+        description="Array of profile steps with target temperature and temperature hold time.",
+    )
+    blockMaxVolumeUl: float = Field(
+        ...,
+        description="Amount of liquid in uL of the most-full well in labware loaded onto the thermocycler.",
+    )
+
+
+class RunProfileResult(BaseModel):
+    """Result data from running a Thermocycler profile."""
+
+
+class RunProfileImpl(AbstractCommandImpl[RunProfileParams, RunProfileResult]):
+    """Execution implementation of a Thermocycler's run profile command."""
+
+    def __init__(
+        self,
+        state_view: StateView,
+        equipment: EquipmentHandler,
+        **unused_dependencies: object,
+    ) -> None:
+        self._state_view = state_view
+        self._equipment = equipment
+
+    async def execute(self, params: RunProfileParams) -> RunProfileResult:
+        """Run a Thermocycler profile."""
+        thermocycler_state = self._state_view.modules.get_thermocycler_module_substate(
+            params.moduleId
+        )
+        thermocycler_hardware = self._equipment.get_module_hardware_api(
+            thermocycler_state.module_id
+        )
+
+        steps = []
+        for profile_step in params.profile:
+            target_temperature = thermocycler_state.validate_target_block_temperature(
+                profile_step.celsius
+            )
+            steps.append(
+                {
+                    "temperature": target_temperature,
+                    "hold_time_seconds": profile_step.holdSeconds,
+                }
+            )
```

Comment:
I think I might slightly prefer a for-comprehension here

```suggestion
        steps = [
            {
                "temperature": thermocycler_state.validate_target_block_temperature(
                    profile_step.celsius
                ),
                "hold_time_seconds": profile_step.holdSeconds,
            }
            for profile_step in params.profile   
        ]
```